src/vcu/src/controls.c

```diff
@@ -47,6 +57,7 @@ bool get_controls_enabled(void) {
 
 void execute_controls(void) {
```

Comment:
Not paramount, but this function is really long and complicated. Can it be broken up into smaller functions? Also can the logic be structured to use less nested if / switch statements / conditionals (&&, !, ||)?